src/Core/StaticConfigTrait.php

```diff
@@ -48,50 +48,41 @@
      * Assuming that the class' name is `Cache` the following scenarios
      * are supported:
      *
-     * Reading config data back:
-     *
-     * ```
-     * Cache::config('default');
-     * ```
-     *
      * Setting a cache engine up.
      *
      * ```
-     * Cache::config('default', $settings);
+     * Cache::setConfig('default', $settings);
      * ```
      *
      * Injecting a constructed adapter in:
      *
      * ```
-     * Cache::config('default', $instance);
+     * Cache::setConfig('default', $instance);
      * ```
      *
      * Configure multiple adapters at once:
      *
      * ```
-     * Cache::config($arrayOfConfig);
+     * Cache::setConfig($arrayOfConfig);
      * ```
      *
      * @param string|array $key The name of the configuration, or an array of multiple configs.
-     * @param array|null $config An array of name => configuration data for adapter.
-     * @return array|null Null when adding configuration or an array of configuration data when reading.
+     * @param array $config An array of name => configuration data for adapter.
      * @throws \BadMethodCallException When trying to modify an existing config.
+     * @throws \LogicException When trying to store an invalid structured config array.
+     * @return void
      */
-    public static function config($key, $config = null)
+    public static function setConfig($key, $config = null)
     {
         if ($config === null) {
-            // Read config.
-            if (is_string($key)) {
-                return isset(static::$_config[$key]) ? static::$_config[$key] : null;
+            if (!is_array($key)) {
+                throw new LogicException('If config is null, key must be an array.');
```

Comment:
I didn't mean the exception per se. That made me wonder as well. InvalidArgumentException would work. But As for the suggestion to split it up into setConfig and setConfigArray I am on the non-split up camp.